packages/block-library/src/navigation/map-menu-items-to-blocks.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * WordPress dependencies
+ */
+import { createBlock, parse } from '@wordpress/blocks';
+
+/**
+ * Convert block attributes to menu item.
+ *
+ * For more documentation on the individual fields present on a menu item please see:
+ * https://core.trac.wordpress.org/browser/tags/5.7.1/src/wp-includes/nav-menu.php#L789
+ *
+ * @param {Object} menuItem the menu item to be converted to block attributes.
+ * @param {Object} menuItem.title stores the raw and rendered versions of the title/label for this menu item.
+ * @param {Array} menuItem.xfn the XFN relationships expressed in the link of this menu item.
+ * @param {Array} menuItem.classes the HTML class attributes for this menu item.
+ * @param {string} menuItem.attr_title the HTML title attribute for this menu item.
+ * @param {string} menuItem.object The type of object originally represented, such as 'category', 'post', or 'attachment'.
+ * @param {string} menuItem.object_id The DB ID of the original object this menu item represents, e.g. ID for posts and term_id for categories.
+ * @param {string} menuItem.description The description of this menu item.
+ * @param {string} menuItem.url The URL to which this menu item points.
+ * @param {string} menuItem.type The family of objects originally represented, such as 'post_type' or 'taxonomy'.
+ * @param {string} menuItem.target The target attribute of the link element for this menu item.
+ * @return {Object} the block attributes converted from the menu item.
+ */
+export const menuItemToBlockAttributes = ( {
+	title: menuItemTitleField,
+	xfn,
+	classes,
+	// eslint-disable-next-line camelcase
+	attr_title,
+	object,
+	// eslint-disable-next-line camelcase
+	object_id,
+	description,
+	url,
+	type: menuItemTypeField,
+	target,
+} ) => {
+	return {
+		label: menuItemTitleField.rendered,
+		rel: xfn.join( ' ' ).trim(),
+		className: classes.join( ' ' ).trim(),
```

Comment:
I was looking at block attrs <--> menu item fields conversion. I couldn't decide whether we should:

1. Set a default value at the point of conversion.
2. Omit missing values on the source object on the target and allow them to fall through to whatever defaults are provided.

I've gone with option 2 because often the defaults are set by Core and thus we shouldn't set our own duplicate defaults here. 

For example block attrs -> menu item. [The menu item code in Core actually has a tonne of defaults](https://core.trac.wordpress.org/browser/tags/5.7.1/src/wp-includes/nav-menu.php#L438) so we don’t need to pass these. In fact there’s a good argument we should allow Core to define the defaults and just pass anything the block has overridden.

I've updated the code and associated tests to reflect this.